File: Threads Posix/exo1.c
#include <stdio.h>
#include <pthread.h>


void* afficher_tableau(void* arg) 
{
    int* tab = (int*)arg;
    
    printf("Contenu du tableau dans le thread : ");
    for (int i = 0; i < 5; i++) 
    {
        printf("%d ", tab[i]);
    }
    printf("\n");
    
    return NULL;
}

int main() 
{
    pthread_t thread;
    int mon_tableau[] = {10, 20, 30, 40, 50};

   
    pthread_create(&thread, NULL, afficher_tableau, (void*)mon_tableau);

    pthread_join(thread, NULL);

    return 0;
}

File: Threads Posix/exo1.py
import threading

def afficher_tableau(tab):
    print("Contenu du tableau dans le thread : ", end="")
    for val in tab:
        print(f"{val} ", end="")
    print()

def main():
    mon_tableau = [10, 20, 30, 40, 50]

    thread = threading.Thread(target=afficher_tableau, args=(mon_tableau,))
    
    thread.start()
    thread.join()

if __name__ == "__main__":
    main()